pkg/tenant/invites_test.go

```diff
@@ -0,0 +1,177 @@
+	userID := newClaims.ParseUserID()
+	require.NotEmpty(userID, "expected user id to be set in access token fixture")
+
+	member := &db.Member{
+		OrgID: orgID,
+		ID:    userID,
+		Email: "kate@example.co",
+		Role:  perms.RoleMember,
+	}
+
+	data, err := member.MarshalValue()
+	require.NoError(err, "could not marshal member data")
+
+	key, err := member.Key()
+	require.NoError(err, "could not create member key from fixture")
+
+	// trtl should return the member data on get
+	trtl.OnGet = func(ctx context.Context, gr *pb.GetRequest) (*pb.GetReply, error) {
+		switch gr.Namespace {
+		case db.MembersNamespace:
+			switch {
+			case bytes.Equal(gr.Key, key):
+				return &pb.GetReply{
+					Value: data,
+				}, nil
+			default:
+				return nil, status.Errorf(codes.NotFound, "member not found")
+			}
+		default:
+			return nil, errors.New("unexpected namespace")
+		}
+	}
+
+	// trtl should return OK on put
+	trtl.OnPut = func(ctx context.Context, pr *pb.PutRequest) (_ *pb.PutReply, err error) {
+		switch pr.Namespace {
+		case db.MembersNamespace:
+			switch {
+			case bytes.Equal(pr.Key, key):
+				// Ensure that the timestamps are updated
+				var newMember db.Member
+				if err = newMember.UnmarshalValue(pr.Value); err != nil {
+					return nil, err
+				}
+
+				if newMember.JoinedAt.IsZero() {
+					return nil, errors.New("joined at timestamp not set for invited member")
+				}
+
+				if newMember.LastActivity.IsZero() {
+					return nil, errors.New("last activity timestamp not set for invited member")
+				}
+				return &pb.PutReply{}, nil
+			default:
+				return nil, status.Errorf(codes.NotFound, "member not found")
+			}
+		default:
+			return nil, errors.New("unexpected namespace")
+		}
+	}
+
+	// Initial Quarterdeck mock returns auth credentials
+	creds := &qd.LoginReply{
+		AccessToken:  accessToken,
+		RefreshToken: refreshToken,
+	}
+	s.quarterdeck.OnInvitesAccept(mock.UseStatus(http.StatusOK), mock.UseJSONFixture(creds), mock.RequireAuth())
+
+	// Set the initial claims fixture
+	claims := &tokens.Claims{
+		Name:  "Leopold Wentzel",
+		Email: "leopold.wentzel@gmail.com",
+	}
+
+	// Endpoint must be authenticated
+	req := &api.MemberInviteToken{}
+	require.NoError(s.SetClientCSRFProtection(), "could not set csrf protection")
+	err = s.client.InviteAccept(ctx, req)
+	s.requireError(err, http.StatusUnauthorized, "this endpoint requires authentication", "expected error when user is not authenticated")
+
+	// Should error if the token is missing
+	require.NoError(s.SetClientCredentials(claims), "could not set client credentials")
+	err = s.client.InviteAccept(ctx, req)
+	s.requireError(err, http.StatusBadRequest, responses.ErrTryLoginAgain, "expected error when token is missing")
+
+	// Successful invite accept request
+	req.Token = "token1234"
+	err = s.client.InviteAccept(ctx, req)
+	require.NoError(err, "could not accept invite")
+	token, err := s.GetClientAccessToken()
+	require.NoError(err, "could not get access cookie token")
```

Comment:
This is failing right now, I'm trying to figure out why.

Reply:
@pdeziel ok, here is why the tests are failing: the max age on the cookies is 0 because the max age is computed from the access and refresh tokens expiration. When the max age is 0 or negative, the cookie jar deletes the cookie instead of setting it. Instead of using constant token strings, we need to use strings that have an expiration in the future for the test to pass.